pkg/ddl/job_table.go

```diff
@@ -184,7 +184,10 @@ func (s *jobScheduler) close() {
 	}
 }
 
+// getJob reads tidb_ddl_job and returns the first runnable DDL job.
 func (s *jobScheduler) getJob(se *sess.Session, tp jobType) (*model.Job, error) {
```

Comment:
we need merge 2 `loadDDLJobAndRun` into 1 to hand a 'fair' case below:
- job 1: add column to test.t
- job 2: truncate test.t: skipped scan since job 1 is running
- job 3: add index to test.t: before we check it, job 1 finished, so it can be run, but it should be run after job 2

since it's already a large pr, i add a task item to track and we can change it later, and can do it together with remove the sub-query inside getJobSQL to speedup.